lib/capybara/cuprite/browser.rb

```diff
@@ -257,6 +268,21 @@ def attach_page(target_id = nil)
         target.page = Page.new(target.client, context_id: target.context_id, target_id: target.id)
         target.page
       end
+
+      def frame_obscured_at?(x, y)
+        frame = page.active_frame
+        return false if frame.main?
+
+        frame_node = frame.evaluate("window.frameElement")
+        return false unless frame_node
+
+        switch_to_frame(:parent)
+        begin
+          obscured?(frame_node, x, y)
```

Comment:
I think it might be possible to check it even without switching frames, but not 100%